tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,232 @@
+import dataclasses
+import uuid
+from string import Template
+from typing import Optional
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.steps.provision.testcloud import GuestTestcloud
+from tmt.utils import field
+
+DEFAULT_IMAGE_BUILDER = "quay.io/centos-bootc/bootc-image-builder:latest"
+
+
+class GuestBootc(GuestTestcloud):
+    containerimage: str
+
+    def __init__(self,
+                 *,
+                 data: tmt.steps.provision.GuestData,
+                 name: Optional[str] = None,
+                 parent: Optional[tmt.utils.Common] = None,
+                 logger: tmt.log.Logger,
+                 containerimage: Optional[str]) -> None:
+        super().__init__(data=data, logger=logger, parent=parent, name=name)
+
+        if containerimage:
+            self.containerimage = containerimage
+
+    def remove(self) -> None:
+        if self.containerimage:
+            tmt.utils.Command(
+                "podman", "rmi", self.containerimage
+                ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+        super().remove()
+
+
+@dataclasses.dataclass
+class BootcData(tmt.steps.provision.testcloud.ProvisionTestcloudData):
+    containerfile: Optional[str] = field(
+        default=None,
+        option=('--containerfile'),
+        metavar='CONTAINERFILE',
+        help="""
+             Select container file to be used to build a container image
+             that is then used by bootc image builder to create a disk image.
+
+             Cannot be used with containerimage.
+             """)
+
+    containerfile_workdir: str = field(
+        default=".",
+        option=('--containerfile-workdir'),
+        metavar='CONTAINERFILE_WORKDIR',
+        help="""
+             Select working directory for the podman build invocation.
+             """)
+
+    containerimage: Optional[str] = field(
+        default=None,
+        option=('--containerimage'),
+        metavar='CONTAINERIMAGE',
+        help="""
+             Select container image to be used to build a bootc disk.
+             This takes priority over containerfile.
+             """)
+
+    add_deps: bool = field(
+        default=True,
+        is_flag=True,
+        option=('--add-deps'),
+        help="""
+             Add tmt dependencies to the supplied container image or image built
+             from the supplied Containerfile.
+             This will cause a derived image to be built from the supplied image.
+             """)
+
+    image_builder: str = field(
+        default=DEFAULT_IMAGE_BUILDER,
+        option=('--image-builder'),
+        metavar='IMAGEBUILDER',
+        help="""
+             The full repo:tag url of the bootc image builder image to use for
+             building the bootc disk image.
+             """)
+
+    def set_image(self, image: str) -> None:
```

Comment:
This kind of setters is not necessary, `self.image = image` is perfectly acceptable.